e2etest/l2tests/assignment.go

```diff
@@ -678,4 +698,269 @@ var _ = ginkgo.Describe("IP Assignment", func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 	})
+	ginkgo.Context("PREFER DUALSTACK", func() {
+		const v4PoolAddresses = "192.168.10.100/32"
+		const v6PoolAddresses = "fc00:f853:0ccd:e799::/124"
+		const v4PoolAddresses2 = "192.168.11.100/32"
+		const v6PoolAddresses2 = "fc00:f853:0ccd:e800::/124"
+		poolv4 := metallbv1beta1.IPAddressPool{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-ns-ipv4-pool"},
+			Spec: metallbv1beta1.IPAddressPoolSpec{
+				Addresses: []string{
+					v4PoolAddresses,
+				},
+			},
+		}
+		poolv6 := metallbv1beta1.IPAddressPool{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-ns-ipv6-pool"},
+			Spec: metallbv1beta1.IPAddressPoolSpec{
+				Addresses: []string{
+					v6PoolAddresses,
+				},
+			},
+		}
+		poolDual := metallbv1beta1.IPAddressPool{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-ns-dualstack-pool"},
+			Spec: metallbv1beta1.IPAddressPoolSpec{
+				Addresses: []string{
+					v4PoolAddresses2,
+					v6PoolAddresses2,
+				},
+			},
+		}
+		ginkgo.BeforeEach(func() {
+			poolv4.Spec.AllocateTo = &metallbv1beta1.ServiceAllocation{Namespaces: []string{testNamespace}}
```

Comment:
do we need the allocation?

Reply:
I think we do, `testnamespace` is changed for every It, so it's impossible to do the Allocation beforehands. Or maybe there's something I'm missing here?

Reply:
if you don't specify the selector, the pool will apply to all the services. Not sure if I am missing something myself :D

Reply:
Sure, it's just that in our e2e tests, there's always a selector (to the testnamespace) in all services so I didn't want to change that. If that's acceptable I can remove the allocation.